### PR TITLE
Management do not rely on event stream ordering now + Version upgrade (-pre9)

### PR DESCRIPTION
**Pull request by mathieucarbou**
Depends on https://github.com/Terracotta-OSS/terracotta-platform/pull/267

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -50,7 +50,7 @@ ext {
   sizeofVersion = '0.3.0'
 
   // Clustered
-  terracottaPlatformVersion = '5.2.0-pre8'
+  terracottaPlatformVersion = '5.2.0-pre9'
```
Comment (mathieucarbou) [inline build.gradle]: next version of tc-platform to release

File: clustered/server/src/main/java/org/ehcache/clustered/server/management/ClientStateSettingsManagementProvider.java [hunk 2]
```diff
@@ -29,7 +29,7 @@
 import java.util.TreeSet;
 
 @Named("ClientStateSettings")
-@RequiredContext({@Named("consumerId"), @Named("clientId"), @Named("type")})
+@RequiredContext({@Named("consumerId"), @Named("type"), @Named("alias")})
```
Comment (mathieucarbou) [inline clustered/server/src/main/java/org/ehcache/clustered/server/management/ClientStateSettingsManagementProvider.java]: `clientId` was wrongly used - should be alias

File: clustered/server/src/main/java/org/ehcache/clustered/server/management/ClientStateSettingsManagementProvider.java [hunk 3]
```diff
@@ -44,12 +44,7 @@ protected ExposedClientStateBinding internalWrap(Context context, ClientStateBin
   private static class ExposedClientStateBinding extends ExposedClientBinding<ClientStateBinding> {
 
     ExposedClientStateBinding(Context context, ClientStateBinding clientBinding) {
-      super(context, clientBinding);
-    }
-
-    @Override
-    public Context getContext() {
-      return super.getContext().with("type", "ClientState");
+      super(context.with("type", "ClientState"), clientBinding);
```
Comment (mathieucarbou) [inline clustered/server/src/main/java/org/ehcache/clustered/server/management/ClientStateSettingsManagementProvider.java]: no need to override

File: clustered/server/src/main/java/org/ehcache/clustered/server/management/ClientStateSettingsManagementProvider.java [hunk 4]
```diff
@@ -44,12 +44,7 @@ protected ExposedClientStateBinding internalWrap(Context context, ClientStateBin
   private static class ExposedClientStateBinding extends ExposedClientBinding<ClientStateBinding> {
 
     ExposedClientStateBinding(Context context, ClientStateBinding clientBinding) {
-      super(context, clientBinding);
-    }
-
-    @Override
-    public Context getContext() {
-      return super.getContext().with("type", "ClientState");
```
Comment (mathieucarbou) [inline clustered/server/src/main/java/org/ehcache/clustered/server/management/ClientStateSettingsManagementProvider.java]: no need to override

File: clustered/server/src/main/java/org/ehcache/clustered/server/management/ClusteredTierManagerSettingsManagementProvider.java [hunk 5]
```diff
@@ -41,12 +41,7 @@ public ClusteredTierManagerSettingsManagementProvider() {
   private static class ExposedClusteredTierManagerBinding extends ExposedAliasBinding<ClusteredTierManagerBinding> {
 
     public ExposedClusteredTierManagerBinding(final Context context, final ClusteredTierManagerBinding binding) {
-      super(context, binding);
-    }
-
-    @Override
-    public Context getContext() {
-      return super.getContext().with("type", "ClusteredTierManager");
```
Comment (mathieucarbou) [inline clustered/server/src/main/java/org/ehcache/clustered/server/management/ClusteredTierManagerSettingsManagementProvider.java]: no need to override

File: clustered/server/src/main/java/org/ehcache/clustered/server/management/Management.java [hunk 6]
```diff
@@ -67,17 +61,6 @@ public Management(ServiceRegistry services, EhcacheStateService ehcacheStateServ
...
-        // workaround for https://github.com/Terracotta-OSS/terracotta-core/issues/426
-        EntityEventService entityEventService = Objects.requireNonNull(services.getService(new BasicServiceConfiguration<>(EntityEventService.class)));
-        entityEventService.addEntityEventListener(new EntityEventListenerAdapter() {
-          @Override
-          public void onCreated() {
-            LOGGER.trace("[{}] onCreated()", entityEventService.getConsumerId());
-            init();
-            sharedPoolsConfigured();
-          }
-        });
```
Comment (mathieucarbou) [inline clustered/server/src/main/java/org/ehcache/clustered/server/management/Management.java]: workaround not needed anymore: management calls are internally delayed

File: clustered/server/src/main/java/org/ehcache/clustered/server/management/Management.java [hunk 7]
```diff
@@ -186,15 +169,15 @@ public void storeAttached(ClientDescriptor clientDescriptor, ClientState clientS
     if (managementRegistry != null) {
       LOGGER.trace("storeAttached({}, {})", clientDescriptor, storeName);
       managementRegistry.refresh();
-      managementRegistry.pushServerEntityNotification(new ClientBinding(clientDescriptor, clientState), "EHCACHE_SERVER_STORE_ATTACHED", Context.create("storeName", storeName));
+      managementRegistry.pushServerEntityNotification(new ClientStateBinding(clientDescriptor, clientState), "EHCACHE_SERVER_STORE_ATTACHED", Context.create("storeName", storeName));
```
Comment (mathieucarbou) [inline clustered/server/src/main/java/org/ehcache/clustered/server/management/Management.java]: bugfix: `EHCACHE_SERVER_STORE_ATTACHED ` was not sent

File: clustered/server/src/main/java/org/ehcache/clustered/server/management/Management.java [hunk 8]
```diff
@@ -186,15 +169,15 @@ public void storeAttached(ClientDescriptor clientDescriptor, ClientState clientS
...
     }
   }
 
   public void storeReleased(ClientDescriptor clientDescriptor, ClientState clientState, String storeName) {
     if (managementRegistry != null) {
       LOGGER.trace("storeReleased({}, {})", clientDescriptor, storeName);
       managementRegistry.refresh();
-      managementRegistry.pushServerEntityNotification(new ClientBinding(clientDescriptor, clientState), "EHCACHE_SERVER_STORE_RELEASED", Context.create("storeName", storeName));
+      managementRegistry.pushServerEntityNotification(new ClientStateBinding(clientDescriptor, clientState), "EHCACHE_SERVER_STORE_RELEASED", Context.create("storeName", storeName));
```
Comment (mathieucarbou) [inline clustered/server/src/main/java/org/ehcache/clustered/server/management/Management.java]: bugfix: `EHCACHE_SERVER_STORE_RELEASED ` was not sent

File: clustered/server/src/main/java/org/ehcache/clustered/server/management/PoolSettingsManagementProvider.java [hunk 9]
```diff
@@ -50,12 +50,7 @@ protected ExposedPoolBinding internalWrap(Context context, PoolBinding managedOb
   private static class ExposedPoolBinding extends ExposedAliasBinding<PoolBinding> {
 
     ExposedPoolBinding(Context context, PoolBinding binding) {
-      super(context, binding);
-    }
-
-    @Override
-    public Context getContext() {
-      return super.getContext().with("type", "Pool");
```
Comment (mathieucarbou) [inline clustered/server/src/main/java/org/ehcache/clustered/server/management/PoolSettingsManagementProvider.java]: no need to override

File: clustered/server/src/main/java/org/ehcache/clustered/server/management/PoolStatisticsManagementProvider.java [hunk 10]
```diff
@@ -75,15 +75,10 @@ protected StatisticRegistry getStatisticRegistry(PoolBinding managedObject) {
...
 
     PoolExposedStatistics(Context context, PoolBinding binding, StatisticRegistry statisticRegistry) {
-      super(context, binding, statisticRegistry);
+      super(context.with("type", "Pool"), binding, statisticRegistry);
       getRegistry().registerSize("AllocatedSize", descriptor("allocatedSize", tags("tier", "Pool")));
     }
 
-    @Override
-    public Context getContext() {
-      return super.getContext().with("type", "Pool");
```
Comment (mathieucarbou) [inline clustered/server/src/main/java/org/ehcache/clustered/server/management/PoolStatisticsManagementProvider.java]: no need to override

File: clustered/server/src/main/java/org/ehcache/clustered/server/management/ServerStoreSettingsManagementProvider.java [hunk 11]
```diff
@@ -52,12 +52,7 @@ protected ExposedServerStoreBinding internalWrap(Context context, ServerStoreBin
   private static class ExposedServerStoreBinding extends ExposedAliasBinding<ServerStoreBinding> {
 
     ExposedServerStoreBinding(Context context, ServerStoreBinding binding) {
-      super(context, binding);
-    }
-
-    @Override
-    public Context getContext() {
-      return super.getContext().with("type", "ServerStore");
```
Comment (mathieucarbou) [inline clustered/server/src/main/java/org/ehcache/clustered/server/management/ServerStoreSettingsManagementProvider.java]: no need to override

File: clustered/server/src/main/java/org/ehcache/clustered/server/management/ServerStoreStatisticsManagementProvider.java [hunk 12]
```diff
@@ -61,11 +61,6 @@
       getRegistry().registerSize("TableCapacity", descriptor("tableCapacity", tags("tier", "Store")));
     }
 
-    @Override
-    public Context getContext() {
-      return super.getContext().with("type", "ServerStore");
```
Comment (mathieucarbou) [inline clustered/server/src/main/java/org/ehcache/clustered/server/management/ServerStoreStatisticsManagementProvider.java]: no need to override

File: management/src/main/java/org/ehcache/management/providers/EhcacheStatisticCollectorProvider.java [hunk 13]
```diff
@@ -18,12 +18,11 @@
...
 import org.terracotta.management.registry.Named;
 import org.terracotta.management.registry.RequiredContext;
-import org.terracotta.management.registry.collect.StatisticCollector;
 import org.terracotta.management.registry.collect.StatisticCollectorProvider;
 
 @RequiredContext(@Named("cacheManagerName"))
-public class EhcacheStatisticCollectorProvider extends StatisticCollectorProvider<StatisticCollector> {
+public class EhcacheStatisticCollectorProvider extends StatisticCollectorProvider {
   public EhcacheStatisticCollectorProvider(ManagementRegistryServiceConfiguration configuration) {
-    super(StatisticCollector.class, configuration.getContext());
```
Comment (mathieucarbou) [inline management/src/main/java/org/ehcache/management/providers/EhcacheStatisticCollectorProvider.java]: little fix: `StatisticCollectorProvider` does not need to be paramaterized

File: build.gradle (hunk 1 above)
Comment (anthonydahanne) [inline build.gradle]: should be available in 10 mn or so

**Comment by anthonydahanne** [comment]
test this please

**Comment by anthonydahanne** [comment]
too soon
test this please

**Comment by mathieucarbou** [comment]
Fixing it.. build failed because of https://github.com/Terracotta-OSS/terracotta-platform/pull/268

**Comment by mathieucarbou** [comment]
@anthonydahanne 👍 
PR merged: https://github.com/Terracotta-OSS/terracotta-platform/pull/270, -pre10 released. 
Build running...

**Comment by anthonydahanne** [comment]
test this please

**Comment by mathieucarbou** [comment]
test this please

**Comment by mathieucarbou** [comment]
build restarted 2 times because of unstable tetss

**Pull request merged**